### PR TITLE
Cleanup storage

### LINKED ISSUE #819: Cleanup initial-storage
- [x] Squash all commits
- [x] Go over the storage_api and write a consistent api, and I agree that a get_ prefix might be nicer to read.
- [x] misfit_expected in test
- [x] Cleanup logging

### PR DESCRIPTION
**Pull request by sondreso**
**Issue**
Resolves #819

### CODE REVIEW

**Comment by ManInFez** [comment]
should we move the api documentation into the doc folder?

**Comment by sondreso** [comment]
Yes, I think we should! But I would also like to test the API against the specification (maybe something like https://pypi.org/project/swagger-tester/), so that they don't get out of sync.

**Comment by ManInFez** [comment]
Yes that is probably a good idea

**Comment by oysteoh** [comment]
just a small comment - we should add flask as a install requirement

**Comment by oysteoh** [comment]
blacking of the files would also be nice - some whitespace / new lines missing etc...

**Comment by sondreso** [comment]
@oysteoh Now things should be fixed, got hit by an upgrade in mock that does not work with python2 all of a sudden :man_shrugging:

File: ert_logging/logger.conf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    formatter: simple
+    filename: api-log.txt
+loggers:
+  ert_shared.storage:
+    level: DEBUG
+    handlers: [apifile]
+    propagate: no
+root:
+  level: DEBUG
+  handlers: [console, file]
```
Comment (jondequinor) [inline ert_logging/logger.conf:28]: missing newline

File: ert_logging/logger.conf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+version: 1
```
Comment (jondequinor) [inline ert_logging/logger.conf:1]: This logger is disjoint from all other loggers in `libres`?

Comment (ManInFez) [reply ert_logging/logger.conf:1]: The only thing worth mentioning in libres is the ert_logger module, which is the one posting to devnull. As for anything other logger in libres, they are coming from the C layer, which is not something we wish to deal with. This is just configuring the default python logging module. If the config is loaded prior to using logging module, you will get that config. There is no other proper configuration of the logging module in either libres nor ert.

**Pull request merged**